jetbrains-core/src/software/aws/toolkits/jetbrains/ui/RegionSelector.kt

```diff
@@ -44,9 +42,21 @@ class RegionSelector : ComboBox<AwsRegion>() {
             }
         }
 
-    private companion object {
-        val RENDERER = SimpleListCellRenderer.create<AwsRegion>("") {
+    companion object {
+        private val RENDERER = SimpleListCellRenderer.create<AwsRegion>("") {
             it.displayName
         }
+
+        fun Cell.regionSelector(prop: KMutableProperty0<AwsRegion>, serviceId: String? = null): CellBuilder<ComboBox<AwsRegion>> {
+            val regionProvider = AwsRegionProvider.getInstance()
+            val regions = when {
+                serviceId != null -> regionProvider.allRegionsForService(serviceId).values.toMutableList()
+                else -> regionProvider.allRegions().values.toMutableList()
+            }
+            val model = CollectionComboBoxModel(regions)
+            return comboBox(model, prop, RENDERER).applyToComponent {
```

Comment:
is this preferred over reusing our own component?

Reply:
I don't think we need to. long term we should move off .form files and so we shouldn't need our own